subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -679,6 +679,59 @@ static inline int time_left(uint32_t start, uint32_t timeout)
 	return timeout - elapsed;
 }
 
+static int wait(int sock, int timeout, int event)
+{
+	struct zsock_pollfd fds = {
+		.fd = sock,
+		.events = event,
+	};
+	int ret;
+
+	ret = zsock_poll(&fds, 1, timeout);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (ret == 1) {
+		if (fds.revents & ZSOCK_POLLNVAL) {
+			return -EBADF;
+		}
+
+		if (fds.revents & ZSOCK_POLLERR) {
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int wait_for_reason(int sock, int timeout, int reason)
+{
+	if (reason == MBEDTLS_ERR_SSL_WANT_READ) {
+		return wait(sock, timeout, ZSOCK_POLLIN);
+	}
+
+	if (reason == MBEDTLS_ERR_SSL_WANT_WRITE) {
+		return wait(sock, timeout, ZSOCK_POLLOUT);
+	}
+
+	/* Any other reason - no way to monitor, just wait for some time. */
+	k_msleep(100);
```

Comment:
I don't really think it'd be useful to have this configurable, it's just some arbitrary value and I'd rather avoid multiplying unnecessary configs. I can make a define out of it though, so that in case it turns out there is a use case for modifying the value, we could easily switch to Kconfig.